sites/public/src/ListingView.tsx

```diff
@@ -296,37 +295,41 @@ export const ListingView = (props: ListingProps) => {
 
   // Move the above methods into our shared helper library when setup
 
+  const getDateString = (date: Date, format: string) => {
+    return date ? dayjs(date).format(format) : null
+  }
+
   const applySidebar = () => (
     <>
       <GetApplication
         onlineApplicationURL={getOnlineApplicationURL()}
-        applicationsDueDate={dayjs(listing.applicationDueDate).format(
-          `MMM. DD, YYYY [${t("t.at")}] h A`
-        )}
         applicationsOpen={!appOpenInFuture}
-        applicationsOpenDate={dayjs(listing.applicationOpenDate).format("MMMM D, YYYY")}
+        applicationsOpenDate={getDateString(listing.applicationOpenDate, "MMMM D, YYYY")}
         paperApplications={getPaperApplications()}
         paperMethod={!!getMethod(listing.applicationMethods, ApplicationMethodType.FileDownload)}
-        postmarkedApplicationsReceivedByDate={dayjs(
-          listing.postmarkedApplicationsReceivedByDate
-        ).format(`MMM. DD, YYYY [${t("t.at")}] h A`)}
+        postmarkedApplicationsReceivedByDate={getDateString(
+          listing.postmarkedApplicationsReceivedByDate,
+          `MMM. DD, YYYY [${t("t.at")}] hh:mm A`
+        )}
         applicationPickUpAddressOfficeHours={listing.applicationPickUpAddressOfficeHours}
         applicationPickUpAddress={getAddress(listing.applicationPickUpAddressType, "pickUp")}
         preview={props.preview}
         listingStatus={listing.status}
       />
       <SubmitApplication
-        applicationMailingAddress={listing.applicationMailingAddress}
+        applicationMailingAddress={getAddress(listing.applicationMailingAddressType, "mailIn")}
         applicationDropOffAddress={getAddress(listing.applicationDropOffAddressType, "dropOff")}
         applicationDropOffAddressOfficeHours={listing.applicationDropOffAddressOfficeHours}
         applicationOrganization={listing.applicationOrganization}
         postmarkedApplicationData={{
-          postmarkedApplicationsReceivedByDate: dayjs(
-            listing.postmarkedApplicationsReceivedByDate
-          ).format(`MMM. DD, YYYY [${t("t.at")}] h A`),
+          postmarkedApplicationsReceivedByDate: getDateString(
+            listing.postmarkedApplicationsReceivedByDate,
+            `MMM. DD, YYYY [${t("t.at")}] hh:mm A`
+          ),
           developer: listing.developer,
-          applicationsDueDate: dayjs(listing.applicationDueDate).format(
-            `MMM. DD, YYYY [${t("t.at")}] h A`
+          applicationsDueDate: getDateString(
+            listing.applicationDueDate,
+            `MMM. DD, YYYY [${t("t.at")}] hh:mm A`
```

Comment:
The formatting of the month doesn't quite work with May, since it's the only one that isn't abbreviated. I'm not sure if this is a problem since it was already like this, but can we either change the formatting so the period isn't necessary or add logic so that it doesn't appear for May?